mlir/test/Quantum/GateBuidlerTest.mlir

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// RUN: quantum-opt --test-gate-builder --split-input-file -verify-diagnostics %s | FileCheck %s
+
+
+%_ = quantum.alloc( 2) : !quantum.reg
+%IN = quantum.extract %_[ 0] : !quantum.reg -> !quantum.bit
+
+// CHECK: [[IN:%.+]] = quantum.extract {{%.+}}[ 0] : !quantum.reg -> !quantum.bit
+// CHECK:  {{%.+}} = quantum.custom "PauliZ"() [[IN]] : !quantum.bit
+// CHECK:  {{%.+}} = quantum.custom "PauliY"() [[IN]] {adjoint} : !quantum.bit
+
+
```

Comment:
Are you still fighting with this? 😆

Reply:
Yes, I don't plan for this to cover all cases now, we can add more builders as we encounter the needs. I think just getting this primitive one in would already be very helpful.

And yes, it was an intense battle with tablegen

Reply:
(but I tamed it in the end)

Reply:
Sorry was talking about the excess empty lines at the end of the file 😅

Reply:
> Sorry was talking about the excess empty lines at the end of the file 😅

:laughing: I actually got sublime to work! But it only has a "ensure eof new line on save", no ensuring only one new line...

(I also found the trim trailing white space)

Reply:
New lines are whitespace so it should be covered under "trailing whitespace" :)

Reply:
> New lines are whitespace so it should be covered under "trailing whitespace" :)

I guess sublime isn't smart enough here :(
(Probably because I'm not paying them?)